Initial commit: efax-gtk

wberrier opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. https://sourceforge.net/p/efax-gtk/discussion/377241/thread/5436870976/
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

TODO:

* [x] send appdata upstream
* [x] figure out dbus permissions
* [x] find/create 128x128 icon (unless there's a way to disable this check?) -> ended up scaling the icon, ugh
* [x] test printing
* [x] default viewer (`xdg-open`)

hfiguiere commented on net.efaxgtk.efaxgtk.yaml:
```diff
@@ -0,0 +1,70 @@
+  - "/share/doc"
+modules:
+  # Shamelessly stole tiff and ghostscript modules from gimp (and converted to yaml)
+  - name: libtiff
```
libtiff is in the runtime. isn't it enough?

wberrier replied:
Hrm, I'll try it out.

wberrier replied:
Ok, yup, that seemed to work, thanks!

hfiguiere commented on net.efaxgtk.efaxgtk.yaml:
```diff
@@ -0,0 +1,70 @@
+app-id: net.efaxgtk.efaxgtk
```
efaxgtk.net doesn't exist.

wberrier replied:
Ah, I didn't realize these represented actual hostnames :)

The website is at: http://efax-gtk.sourceforge.net/

net.sourceforge.efax-gtk?

Do I rename everything but keep using this pull request?

hfiguiere replied:
yep.

wberrier replied:
I used net.sourceforge.efaxgtk.  Let me know if there's a more appropriate name.

hfiguiere commented on net.efaxgtk.efaxgtk.yaml:
```diff
@@ -0,0 +1,70 @@
+  - "/include"
+  - "*.la"
+  - "*.a"
+  - "/lib/debug"
```
lib debug shouldn't be removed. it will be automatically put into the debug package.

hfiguiere commented on net.efaxgtk.efaxgtk.yaml:
```diff
@@ -0,0 +1,70 @@
+      - install -Dp -m 644 efax-gtk.png --target-directory=/app/share/icons/hicolor/48x48/apps
+    sources:
+      - type: archive
+        url: http://prdownloads.sourceforge.net/efax-gtk/efax-gtk-3.2.15.src.tgz
```
please use https url

hfiguiere commented on net.efaxgtk.efaxgtk.yaml:
```diff
@@ -0,0 +1,70 @@
+  - name: c++-gtk-utils
+    sources:
+      - type: archive
+        url: http://prdownloads.sourceforge.net/cxx-gtk-utils/c++-gtk-utils-2.2.20.tar.gz
```
please use http url

hfiguiere commented:
bot, build net.efaxgtk.efaxgtk

wberrier commented:
Hrm, I don't think there's a 128x128 icon available...
```
zgrep "<icon type=\"remote\">" builddir/*/share/app-info/xmls/net.efaxgtk.efaxgtk.xml.gz || test -f builddir/*/share/app-info/icons/flatpak/128x128/net.efaxgtk.efaxgtk.png
```

wberrier commented:
bot, build net.efaxgtk.efaxgtk

hfiguiere commented:
bot, build net.sourceforge.efaxgtk

hfiguiere commented on net.sourceforge.efaxgtk.yaml:
```diff
@@ -0,0 +1,63 @@
+  - --socket=x11
+  - --socket=session-bus
+  - --share=network
+  - --filesystem=host
```
so you need the whole filesystem access?

wberrier replied:
Um, probably not... likely just documents?  Let me look into that...

wberrier replied:
I think it'll at least need `home` (the config file is `~/.efaxgtkrc`)

hfiguiere replied:
you should use `--persist=.efaxgtkrc` for that, and `--filesystem=xdg-documents` for the docments

wberrier replied:
I'm trying this:

```patch
diff --git a/net.sourceforge.efaxgtk.yaml b/net.sourceforge.efaxgtk.yaml
index 7fc0c0d..59d10c7 100644
--- a/net.sourceforge.efaxgtk.yaml
+++ b/net.sourceforge.efaxgtk.yaml
@@ -10,8 +10,9 @@ finish-args:
   - --socket=x11
   - --socket=session-bus
   - --share=network
-  # Looks like the app uses ~/.efax-gtkrc
-  - --filesystem=home
+  - --persist=.efax-gtkrc
+  - --persist=.efax-gtk_mainwin_save
+  - --filesystem=xdg-documents
   # For file browse dialogs
   - --filesystem=xdg-run/gvfsd
   # For modems

```

And I'm getting:

```
Can't open file/home/wberrier/.efax-gtkrc
```

Not quite sure why that's not working... plus, I think it's nice if the user has a pre-existing configuration that it gets picked up with `home`.

hfiguiere commented on net.sourceforge.efaxgtk.yaml:
```diff
@@ -0,0 +1,63 @@
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=session-bus
```
more granular dbus permissions are needed.

hfiguiere replied:
You need at least `--talk-name=org.gtk.vfs.*` for GVFS, and also `--filesystem=xdg-run/gvfs`

hfiguiere commented on net.sourceforge.efaxgtk.yaml:
```diff
@@ -0,0 +1,63 @@
+app-id: net.sourceforge.efaxgtk
+runtime: org.gnome.Platform
```
do you really need the gnome platform?

Freedesktop SDK contains Gtk3.

wberrier replied:
I'll give it a shot.

wberrier replied:
That worked, thanks!

wberrier commented:
Here's the session bus log, nothing really popping out to me about what may be wrong:

```
$ flatpak run --log-session-bus net.sourceforge.efaxgtk
Gtk-Message: 20:01:50.894: Failed to load module "xapp-gtk3-module"
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for net.sourceforge.efaxgtk
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.net.sourceforge.efaxgtk
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C10
B-1: <- org.freedesktop.DBus return from C11
*REWRITTEN*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B-1: <- org.freedesktop.DBus return from C12
*REWRITTEN*
C13: -> org.gtk.vfs.Daemon call org.gtk.vfs.MountTracker.ListMountableInfo at /org/gtk/vfs/mounttracker
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C13
*REWRITTEN*
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C14
Gtk-Message: 20:01:50.923: Failed to load module "canberra-gtk-module"
Gtk-Message: 20:01:50.923: Failed to load module "pk-gtk-module"
Gtk-Message: 20:01:50.924: Failed to load module "canberra-gtk-module"
Gtk-Message: 20:01:50.924: Failed to load module "pk-gtk-module"
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for net.sourceforge.efaxgtk
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.net.sourceforge.efaxgtk
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C10
B-1: <- org.freedesktop.DBus return from C11
B-1: <- org.freedesktop.DBus return from C12
B-1: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.ServiceUnknown from C13
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C14
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/DBus
*DENIED* (ping)
B-1: <- org.freedesktop.DBus return from C15
*REWRITTEN*
C16: -> org.freedesktop.DBus call org.freedesktop.DBus.RequestName at /org/freedesktop/DBus
Filtering message due to arg0 org.cgu.progs.efax_gtk, policy: 0 (required 3)
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C16
*REWRITTEN*

** (efax-gtk:2): CRITICAL **: 20:01:50.927: Dbus error: org.freedesktop.DBus.Error.ServiceUnknown

** (efax-gtk:2): CRITICAL **: 20:01:50.927: Failed to obtain name org.cgu.progs.efax_gtk on session bus: org.freedesktop.DBus.Error.ServiceUnknown

** (efax-gtk:2): WARNING **: 20:01:50.927: Error in present_instance(): either register_prog() has not been called,
there was an error in obtaining a connection or service name in register_prog(),
present_instance() has been called in the first invocation of the program
or present_instance() has already been called once before in this invocation of
the program.  Ignoring this call() to present_instance

C17: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C18: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C19: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C17
B-1: <- org.freedesktop.DBus return from C18
B-1: <- org.freedesktop.DBus return from C19

```

Help?

hfiguiere commented:
I think you need `--own-name=org.cgu.progs.efax_gtk`

wberrier commented:
> I think you need `--own-name=org.cgu.progs.efax_gtk`

That worked, thanks!

hfiguiere commented:
bot, build net.sourceforge.efaxgtk

wberrier commented:
Any advice about launching a postscript viewer?  Meaning, does flatpak have a mechanism to launch other apps based on mimetype?  (That would likely require source changes, I'm assuming...)

It looks like `gv` is used by default from the app, so maybe I can figure out how to bundle and include that.  I'm assuming it's not normal that a flatpak would bundle a viewer like evince?

wberrier commented:
bot, build net.sourceforge.efaxgtk

wberrier commented:
bot, build net.sourceforge.efaxgtk

wberrier commented:
bot, build net.sourceforge.efaxgtk

wberrier commented:
I changed the default config to use `xdg-open` instead of `gv`, as well as printing support.

Any remaining items for this?

barthalion commented:
/merge
